esmvaltool/diag_scripts/climate_patterns/rename_variables.py

```diff
@@ -0,0 +1,232 @@
+# (C) Crown Copyright 2022-2024, Met Office.
+"""Script containing cube re-naming functions for driving scripts.
+
+Author
+------
+Gregory Munday (Met Office, UK)
+"""
+
+
+def rename_clim_variables(cube):
```

Comment:
I feel these functions could be improved to remove the duplication, but I wouldn't want to hold up the merge of this PR for it.

Reply:
I've refactored this file and put it into a single function, inside `sub_functions.py`

Reply:
I've fixed this, and added a new `rename_variables` function to `sub_functions.py`

Reply:
Nice! 🥳